lib/src/Uncertainty/Algorithm/Simulation/CrossEntropyImportanceSampling.cxx

```diff
@@ -0,0 +1,248 @@
+     
+    // Optimize auxiliary distribution parameters
+    auxiliaryDistributionParameters = optimizeAuxiliaryDistributionParameters(auxiliaryCriticInputSamples);
+
+    // Update auxiliary Distribution Parameters
+    updateAuxiliaryDistribution(auxiliaryDistributionParameters); 
+  } // if (getEvent().getOperator()(currentQuantile, getEvent().getThreshold()))
+
+  UnsignedInteger iterationNumber  = 0;
+  
+  while ((getEvent().getOperator()(getEvent().getThreshold(), currentQuantile)) && (currentQuantile != getEvent().getThreshold()))
+  {
+    ++iterationNumber ;
+    Point currentAuxiliaryDistributionParameters = auxiliaryDistributionParameters;
+
+    // Drawing of samples using auxiliary density and evaluation on limit state function
+    auxiliaryInputSample = Sample(0, initialDistribution_.getDimension());
+    auxiliaryOutputSample = Sample(0, 1);
+
+    for (UnsignedInteger i = 0; i < getMaximumOuterSampling(); ++i)
+    {
+      const Sample blockSample(auxiliaryDistribution_.getSample(getBlockSize()));
+      auxiliaryInputSample.add(blockSample);
+      auxiliaryOutputSample.add(computeOutputSamples(blockSample));
+
+      if (stopCallback_.first && stopCallback_.first(stopCallback_.second))
+        throw InternalException(HERE) << "User stopped simulation";
+    }
+
+    // Computation of current quantile
+    currentQuantile = auxiliaryOutputSample.computeQuantile(rhoQuantile_)[0];
+    
+    // If failure probability reached, stop the adaptation
+    if (getEvent().getOperator()(currentQuantile, getEvent().getThreshold()))
+    {
+      currentQuantile = getEvent().getThreshold();
+    }
+    else
+    {
+      Indices indiceCritic(0);
+      for (UnsignedInteger i = 0; i < auxiliaryInputSample.getSize(); ++i)
+      {
+        const Bool weightBool = getEvent().getOperator()(auxiliaryOutputSample(i, 0), currentQuantile);
+        if (weightBool)
+          indiceCritic.add(i);
+      } 
+
+   // Extract the relevant sample
+   const Sample auxiliaryCriticInputSamples(auxiliaryInputSample.select(indiceCritic));
+  
+   // Optimize auxiliary distribution parameters
+   Point auxiliaryDistributionParameters = optimizeAuxiliaryDistributionParameters(auxiliaryCriticInputSamples);
+  
+   // Update auxiliary Distribution Parameters
+   updateAuxiliaryDistribution(auxiliaryDistributionParameters); 
+
+      
+    }
+
+    if (stopCallback_.first && stopCallback_.first(stopCallback_.second))
+      throw InternalException(HERE) << "User stopped simulation";
+
+  } // while
+
+  // Find failure sample indices
+  Indices indicesCritic(0);
+  for (UnsignedInteger i = 0; i < auxiliaryOutputSample.getSize(); ++i)
+  {
+    // Find failure Points
+    if (getEvent().getOperator()(auxiliaryOutputSample(i, 0), getEvent().getThreshold()))
+      indicesCritic.add(i);
+  } // for i
+
+  const Sample resp_sampleCritic(auxiliaryOutputSample.select(indicesCritic));
+  const Sample sampleCritic(auxiliaryInputSample.select(indicesCritic));
```

Comment:
stick to camel case
maybe something with Input/Output in the names, for consistency with auxiliaryOutputSample/auxiliaryInputSample ?

Reply:
fixed.